### PR TITLE
Changes to Prepare for Multiple CC

### PR DESCRIPTION
**Pull request by etwest**
This will need to be a pull request which is synchronized with the changes on the streaming end.

### CODE REVIEW

**Comment by etwest** [comment]
Pull request on the other end coming soon

File: include/buffer_tree.h [hunk 1]
```diff
@@ -122,16 +122,18 @@ class BufferTree {
...
    * Useful when switching from blocking to non-blocking calls
    * to the circular queue
-   * For example: we use this when shutting down the graph_workers\
+   * For example: we set this to true when shutting down the graph_workers
+   * @param    bypass is true if we should turn on non-blocking operations
+   *           and false if we should turn them off
    * @return   nothing
    */
-  void bypass_wait();
+  void bypass_wait(bool bypass);
```
Comment (Victor-C-Zhang) [inline include/buffer_tree.h:130]: since this function now toggles instead of sets, we should rename it to avoid confusion

Comment (etwest) [reply include/buffer_tree.h:130]: Any ideas?

Comment (Victor-C-Zhang) [reply include/buffer_tree.h:130]: `toggle_wait`?

Comment (etwest) [reply include/buffer_tree.h:130]: I think I've going to go with set_non_block(true/false)

**Review by Victor-C-Zhang — APPROVED** [review]

**Pull request merged**